### ISSUE TITLE
'via' field should not be mandatory if scope link is defined

### ISSUE THREAD

**Issue by LarsErikP** [zehweh/puppet-netplan#11]
netplan supports a configuration like this:
```
network:
  version: 2
  renderer: networkd
  ethernets:
    ens3:
      match:
        macaddress: <foo>
      dhcp4: true
      routes:
        - to: 10.212.132.0/24
          scope: 'link'
          table: 1
```
This is not possible with this module, because the "via" field is mandatory. Really not sure what's the best way to fix it, but it should be possible to create a config like this

**Comment by LarsErikP** [comment]
Also, the "scope" setting is missing from the template, so it want be set in the resulting yaml file anyhow.. That's another (related) bug..

**Comment by zehweh** [comment]
Hi,

thanks for your interest in this project.

Weird, according to the netplan docs,  
> at least to and via must be specified

Source: [https://netplan.io/reference#routing](https://netplan.io/reference#routing)

**Comment by LarsErikP** [comment]
That absolutley weird, because the configuration i pasted works fine with netplan.io 0.97-0ubuntu1~18.04.1 on Ubuntu 18.04...

**Comment by Tuxdiver** [comment]
Maybe that changed. Currently the netplan.io website says:

> The routes block defines standard static routes for an interface.
> At least to and via must be specified (except for routes with
> scope: link, where only to is required).

And as LarsErikP mentioned: it works, at least on Ubuntu 20.04 :-)

**Comment by Tuxdiver** [comment]
I created a pull request for this issue: #19

**Comment by zehweh** [comment]
closed by #19
